tiledb/sm/array_schema/dimension.cc

```diff
@@ -2078,6 +2090,59 @@ void Dimension::set_check_range_func() {
   }
 }
 
+void Dimension::set_adjust_range_oob_func() {
+  switch (type_) {
+    case Datatype::INT32:
+      adjust_range_oob_func_ = adjust_range_oob<int32_t>;
+      break;
+    case Datatype::INT64:
+      adjust_range_oob_func_ = adjust_range_oob<int64_t>;
+      break;
+    case Datatype::INT8:
+      adjust_range_oob_func_ = adjust_range_oob<int8_t>;
+      break;
+    case Datatype::UINT8:
+      adjust_range_oob_func_ = adjust_range_oob<uint8_t>;
+      break;
+    case Datatype::INT16:
+      adjust_range_oob_func_ = adjust_range_oob<int16_t>;
+      break;
+    case Datatype::UINT16:
+      adjust_range_oob_func_ = adjust_range_oob<uint16_t>;
+      break;
+    case Datatype::UINT32:
+      adjust_range_oob_func_ = adjust_range_oob<uint32_t>;
+      break;
+    case Datatype::UINT64:
+      adjust_range_oob_func_ = adjust_range_oob<uint64_t>;
+      break;
+    case Datatype::FLOAT32:
+      adjust_range_oob_func_ = adjust_range_oob<float>;
+      break;
+    case Datatype::FLOAT64:
+      adjust_range_oob_func_ = adjust_range_oob<double>;
+      break;
+    case Datatype::DATETIME_YEAR:
+    case Datatype::DATETIME_MONTH:
+    case Datatype::DATETIME_WEEK:
+    case Datatype::DATETIME_DAY:
+    case Datatype::DATETIME_HR:
+    case Datatype::DATETIME_MIN:
+    case Datatype::DATETIME_SEC:
+    case Datatype::DATETIME_MS:
+    case Datatype::DATETIME_US:
+    case Datatype::DATETIME_NS:
+    case Datatype::DATETIME_PS:
+    case Datatype::DATETIME_FS:
+    case Datatype::DATETIME_AS:
```

Comment:
cc @ktsitsi If this PR merges before your `time` patch, please rebase and add support for the `time` type here.

Reply:
Thanks @joe-maley I will have it in mind!